lib/modules/versioning/hermit/index.ts

```diff
@@ -0,0 +1,231 @@
+import { regEx } from '../../../util/regex';
+import { RegExpVersion, RegExpVersioningApi } from '../regex';
+import type { VersioningApiConstructor } from '../types';
+
+export const id = 'hermit';
+export const displayName = 'Hermit';
+export const urls = [
+  'https://cashapp.github.io/hermit/packaging/reference/#versions',
+];
+export const supportsRanges = false;
+
+export class HermitVersioning extends RegExpVersioningApi {
+  static versionRegex =
+    '^(?<major>\\d+)(\\.(?<minor>\\d+))?(\\.(?<patch>\\d+))?(_(?<build>\\d+))?([-]?(?<prerelease>[^.+][^+]*))?([+](?<compatibility>[^.-][^+]*))?$';
+  private _hermitConfig: RegExp | null = null;
+
+  public constructor() {
+    super(HermitVersioning.versionRegex);
+    this._hermitConfig = regEx(HermitVersioning.versionRegex);
+  }
+
+  _parseVersion(version: string): RegExpVersion | null {
```

Comment:
remove, should be possible to test via public API 

if you need to test the parse function directly, move it outside of the class to a common file. but that shouldn't be necessary.